crates/common/download/src/download.rs

```diff
@@ -79,16 +83,115 @@ impl DownloadInfo {
     }
 }
 
+impl<Auth> DownloadInfo<Auth>
+where
+    for<'a> &'a Auth: Into<AnonymisedAuth>,
+{
+    pub fn clone_anonymise_auth(&self) -> DownloadInfo<AnonymisedAuth> {
+        DownloadInfo {
+            url: self.url.clone(),
+            auth: self.auth.as_ref().map(|auth| auth.into()),
+        }
+    }
+}
+
 /// Possible authentication schemes
 #[derive(Debug, Clone, Deserialize, PartialEq, Eq, Serialize)]
 #[serde(rename_all = "camelCase")]
 #[serde(deny_unknown_fields)]
-pub enum Auth {
+pub enum RequiredAuth {
     /// HTTP Bearer authentication
     Bearer(String),
+    /// TLS certificate authentication
+    Certificate,
+}
```

Comment:
I don't understand this change. This is even going in the reverse direction I was expecting.

- In their first version, `tedge-mapper` and `tedge-agent` were exchanging over MQTT the credentials required to download a software package from c8y. This was not only a security issue (the `Bearer` being sent clear-text over MQTT), but also fragile (as the JWT token might expired before being used).
- The introduction of `c8y_auth_proxy` makes the authentication to c8y an internal issue of the mapper. This is less fragile, but still not secure as any local process has full access to the c8y API which is exposed over non-TLS HTTP.
- This PR aims to add HTTPS support with mutual TLS authentication of the mapper and the agent (the agent is possibly running on a different device). Each needs its own certificate (and private key). Each needs to trust the certificate of the other. But *no identity and even no credentials have to be exchanged over MQTT* between the mapper and the agent.
- Hence, *I was expecting the `enum Auth` to be deprecated*, not augmented.